vulcan/lib/client/jsx/components/workflow/session/session_manager.tsx

```diff
@@ -82,15 +85,35 @@ export default function SessionManager() {
     [state.inputs, workflow]
   );
 
+  const hasValidationErrors = useMemo(
```

Comment:
In cases where the computation itself is fairly cheap, it's actually totally ok to just `const hasValidationErrors = Object.keys(state.validationErrors).length > 0`.  This saves complexity in the code as well.  useMemo is definitely most useful when it is important to keep the instance stable or to save against hard computations, but simple predicates like this are also totes fine as a regular old variable :P